index.js

```diff
@@ -119,14 +111,46 @@
   //  $$type :: String
   var $$type = '@@type';
 
-  //  type :: Any -> String
+  //  RPARSE :: RegExp
+  var RPARSE = /^(?:([^/\n\r]+)\/)?(.+?)(?:@(\d+))?$/;
+
+  //       TypeIdentifier :: (a, b, c) -> TypeIdentifier a b c
```

Comment:
Oh - The reason why the signature is like this, is because at this point there are technically *no* restrictions to the input types. It's up to `parseTypeIdentifier` and `parseNativeType` to return a TypeIdentifier of a specific concrete type. Not sure if that's the right way of thinking about it though.